src/graphnet/models/task/task.py

```diff
@@ -380,66 +386,85 @@ def _forward(self, x: Union[Tensor, Data]) -> Tensor:  # type: ignore
 
 
 class StandardFlowTask(Task):
-    """A `Task` for `NormalizingFlow`s in GraphNeT."""
+    """A `Task` for `NormalizingFlow`s in GraphNeT.
+
+    This Task requires the support package`jammy_flows` for constructing and
+    evaluating normalizing flows.
+    """
 
     def __init__(
         self,
-        target_labels: List[str],
+        hidden_size: Union[int, None],
+        flow_layers: str = "gggt",
+        target_norm: float = 1000.0,
         **task_kwargs: Any,
     ):
-        """Construct `StandardLearnedTask`.
+        """Construct `StandardFlowTask`.
 
         Args:
             target_labels: A list of names for the targets of this Task.
-            hidden_size: The number of columns in the output of
-                         the last latent layer of `Model` using this Task.
-                         Available through `Model.nb_outputs`
+            flow_layers: A string indicating the flow layer types. See
+             https://thoglu.github.io/jammy_flows/usage/introduction.html
+             for details.
+            target_norm: A normalization constant used to divide the target
+            values. Value is applied to all targets. Defaults to 1000.
+            hidden_size: The number of columns on which the normalizing flow
+            is conditioned on. May be `None`, indicating non-conditional flow.
         """
         # Base class constructor
-        super().__init__(target_labels=target_labels, **task_kwargs)
 
-    def nb_inputs(self) -> int:
-        """Return number of inputs assumed by task."""
-        return len(self._target_labels)
+        # Member variables
+        self._default_prediction_labels = ["nllh"]
+        self._hidden_size = hidden_size
+        super().__init__(**task_kwargs)
+        self._flow = jammy_flows.pdf(
+            f"e{len(self._target_labels)}",
+            flow_layers,
+            conditional_input_dim=hidden_size,
+        )
+        self._initialized = False
+        self._norm = target_norm
 
-    def _forward(self, x: Tensor, jacobian: Tensor) -> Tensor:  # type: ignore
-        # Leave it as is.
-        return x
+    @property
+    def default_prediction_labels(self) -> List[str]:
+        """Return default prediction labels."""
+        return self._default_prediction_labels
+
+    def nb_inputs(self) -> Union[int, None]:  # type: ignore
+        """Return number of conditional inputs assumed by task."""
```

Comment:
does non-conditional inputs also exist or what is the reason for the distinction of "conditional"?

Reply:
The flow can be trained both as a conditional flow, where it's conditioned on a latent representation of an event e.g. `p(target|latent_rep) `  which is the probability of seeing `target` for event represented by` latent_rep`, and as an unconditional probability distribution e.g. `p(target)` which would just learn the overall distribution of the variable in the dataset.